virtinst/guest.py

```diff
@@ -1206,3 +1207,26 @@ def _remove_duplicate_console(self, dev):
         if condup:
             log.debug("Found duplicate console device:\n%s", condup.get_xml())
             self.devices.remove_child(condup)
+
+    def _remove_spice_audio(self):
+        for audio in self.devices.audio:
+            if audio.type == "spice":
+                self.devices.remove_child(audio)
+
+    def _remove_spice_channels(self):
+        for channel in self.devices.channel:
+            if channel.type == DeviceChannel.TYPE_SPICEVMC:
+                self.devices.remove_child(channel)
+
+    def _remove_spice_usbredir(self):
+        for redirdev in self.devices.redirdev:
+            if redirdev.type == "spicevmc":
+                self.devices.remove_child(redirdev)
+
+    def _remove_spice_devices(self):
+        if self.has_spice():
```

Comment:
This makes me a little paranoid. Can you confirm that there's zero valid config in libvirt that allows spice audio, channel, redirdev if spice graphics isn't configured?

This could also give kinda unexpected results. Say you are using `cat foo.xml | virt-xml --edit --disk path=/foo` but the XML config has incorrect spice devices (maybe the input XML was hand edited). Suddenly the XML diff will show removal of the spice devices too. That isn't wrong, we may be fixing the XML, but it's unexpected IMO

I think just to be really safe here, and future proof incase somehow libvirt validation relaxes in the future, we do something like

```python
def _remove_spice_devices(self, rmdev):
    if (rmdev.DEVICE_TYPE != 'graphics' or
        rmdev.type != 'spice'):
        return
    ...
```

Or maybe we lift that check into `remove_device`, since this function is also reused for `change_graphics`